File: crates/infergen-core/templates/echo/snippet.go
// After `infergen generate`, import your typed SDK:
import infergen "github.com/your-org/your-project/infergen/generated"

func TrackingMiddleware(next echo.HandlerFunc) echo.HandlerFunc {
	return func(c echo.Context) error {
		infergen.Track.HttpRequestHandled(c.Request().Method, c.Path())
		return next(c)
	}
}

func LoginHandler(c echo.Context) error {
	userID := authenticate(c)
	infergen.Track.UserAuthenticated(userID)
	return c.JSON(200, map[string]bool{"ok": true})
}

File: crates/infergen-core/templates/fastapi/snippet.py
# After `infergen generate`, import your typed SDK:
from infergen_generated import track

@app.middleware("http")
async def tracking_middleware(request: Request, call_next):
    track.endpoint_called(path=request.url.path, method=request.method)
    return await call_next(request)

@app.post("/users")
async def register(user: UserCreate):
    new_user = await create_user(user)
    track.user_registered(user_id=str(new_user.id))
    return new_user
